diff --git a/react-native.js b/react-native.js
index 74a8b9e70fde06482b9c9fd4720a497a0895e09f..ca585c9ecaef4885d320bc9cf9084116a2955746 100644
--- a/react-native.js
+++ b/react-native.js
@@ -2230,7 +2230,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
diff --git a/react-native.mjs b/react-native.mjs
index d013dd02d2e0b424e8de026c0576175bcd515f8f..c2fb8c46272f807362c3fb25d96ec00bfa92cc40 100644
--- a/react-native.mjs
+++ b/react-native.mjs
@@ -2209,7 +2209,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
diff --git a/react-native.web.js b/react-native.web.js
index 5db8825d58fc6077bd4f095388dc54ddfdcfd46b..a2d9d389157793ba5511ea2807d64f22a5e52a86 100644
--- a/react-native.web.js
+++ b/react-native.web.js
@@ -2758,7 +2758,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
diff --git a/react-native.web.mjs b/react-native.web.mjs
index 2cdadd061e78b558995593719af06faf7f6baa10..b59d2b74ee7c4e84ad534d584af7c2651904731c 100644
--- a/react-native.web.mjs
+++ b/react-native.web.mjs
@@ -2737,7 +2737,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
diff --git a/react.js b/react.js
index 5db8825d58fc6077bd4f095388dc54ddfdcfd46b..a2d9d389157793ba5511ea2807d64f22a5e52a86 100644
--- a/react.js
+++ b/react.js
@@ -2758,7 +2758,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
diff --git a/react.mjs b/react.mjs
index 2cdadd061e78b558995593719af06faf7f6baa10..b59d2b74ee7c4e84ad534d584af7c2651904731c 100644
--- a/react.mjs
+++ b/react.mjs
@@ -2737,7 +2737,7 @@ function doMaintainScrollAtEnd(ctx) {
             () => {
               state.maintainingScrollAtEnd = false;
             },
-            maintainScrollAtEnd.animated ? 500 : 0
+            maintainScrollAtEnd.animated ? 150 : 0
           );
         } else {
           state.maintainingScrollAtEnd = false;
